components/pageLoader.tsx

```diff
@@ -0,0 +1,114 @@
+import cachedData from "../cachedData.json";
+import Page from "./Page";
+import { JSObjectFromJSON } from "../src/dataProcessing";
+import { getJsonStructure } from "evergreen-org-crawler/src/index"
+import config from "evergreen-org-crawler/config.json"
+import { useEffect, useState } from "react";
+import CircularProgress from '@mui/material/CircularProgress';
+import Box from '@mui/material/Box';
+
+enum Mode{
+	Frontend,
+	StandaloneBackend,
+	IntegratedBackend
+}
+
+//TODO: Move to config file
+const mode: Mode = Mode.IntegratedBackend
+
+function getProperty<T, K extends keyof T>(o: T, propertyName: K): T[K] {
+	return o[propertyName];
+}
+
+async function getDataFromAPI(api: "loadNew" | "loadLatest", request: "npm" | "PyPI" | "RubyGems"){
+	console.log("Start")
+	let JSObject = await fetch("api/" + api)
+	let retries = 10
+	while(!JSObject.ok){
+		JSObject = await fetch("api/" + api)
+		--retries
+		if(retries == 0){
+			throw new Error("Failed to fetch")
+		}
+	}
+	console.log(JSObject)
+	const data = await JSObject.json()
+	const result = JSObjectFromJSON(getProperty(data! as { npm: any, PyPI: any, RubyGems: any }, request))
```

Comment:
As stated in some other comment somewhere else, I don't like this `as`. The api handler file should export some types to be used here in stead of `{npm: any, PyPI: any, RubyHems: any}`